pkg/api/v20160330/validate.go

```diff
@@ -122,7 +122,7 @@ func validatePoolName(poolName string) error {
 }
 
 func validateDNSName(dnsName string) error {
-	dnsNameRegex := `^([a-z][a-z0-9-]{1,45}[a-z0-9])$`
```

Comment:
yes, i overlooked vlab version. 
Just updated, and also correct a little bit to match the error message.